### PR TITLE
Drop the dependency on the Spring Framework

### PR DESCRIPTION
**Pull request by jigga**
1. All Spring dependencies have been removed.
2. OkHttp client is now used to interact with Confluence APIs, and the Authorization header is added via an Interceptor.
3. Jackson is the only JSON processing library.
4. The jvm-test-suite plugin has been applied to separate unit & integration tests.
5. The Gradle publish-plugin has been updated to the latest version (1.2.0).
6. Gradle integration tests for the confluence task first publish the current SNAPSHOT version of the plugin to mavenLocal, so it's no longer necessary to do it manually before running integration tests.
7. Using built-in Java utils for reading and writing files.

### Test evidence

1. Project's README.md on Confluence

image

2. Tree structure when generating pages using the `fileTree`

image

3. All unit tests pass

image

4. All integration tests pass

image

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- OkHttpClient is now used to interact with Confluence REST APIs.
+- Jackson is the only JSON processing library.
+- The jvm-test-suite plugin has been applied to separate unit & integration tests.
+- The Gradle publish-plugin has been updated to the latest version (1.2.0).
+- Gradle's integration tests for the confluence task first publish the current SNAPSHOT version of the plugin to mavenLocal, so it's no longer necessary to do it manually before running integration tests.
+- Using built-in Java utils for reading and writing files.
+- The ``=`` sign is now required when assigning a value to a property. This is because
+  the underlying way the plugin's configuration is applied has changed to make the plugin future-proof.
+  Part of that change was replacing the previously used ``ConfluenceConfig`` class with the ``ConfluenceExtension`` class.
+  The new class also provides the ``pages`` method, but the argument to that method is now
```
Comment (imanushin) [inline RELEASES.md:18]: one more benefit of this is what Action doesn't have compatibility issues with Gradle 9

File: src/integrationTest/java/com/github/qwazer/markdown/confluence/Version.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+public final class Version {
+
+    public static final String PROJECT_VERSION;
+
+    static {
```
Comment (imanushin) [inline src/integrationTest/java/com/github/qwazer/markdown/confluence/Version.java:13]: .properties file can also be parsed with [Properties](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html) class.

File: src/integrationTest/resources/gradle_builds/build_1.gradle [hunk 3]
```diff
@@ -0,0 +1,29 @@
+plugins {
+    id 'com.github.qwazer.markdown-confluence'
+}
+
```
Comment (imanushin) [inline src/integrationTest/resources/gradle_builds/build_1.gradle:4]: Am I right that confluence server must run locally on that port for a proper testing?

**Review by imanushin — APPROVED** [review]

File: RELEASES.md (hunk 1 above)
Comment (jigga) [reply RELEASES.md:18]: That's correct.

File: src/integrationTest/java/com/github/qwazer/markdown/confluence/Version.java (hunk 2 above)
Comment (jigga) [reply src/integrationTest/java/com/github/qwazer/markdown/confluence/Version.java:13]: True, and would probably be much easier this way - let me change that.

File: src/integrationTest/resources/gradle_builds/build_1.gradle (hunk 3 above)
Comment (jigga) [reply src/integrationTest/resources/gradle_builds/build_1.gradle:4]: Yep, but there's an image provided by the author that you can just run and then run the integration tests. By default when running the gradle build, only the unit tests will be executed. To run the integration tests (integrationTest task), you need to run the container and then run the Gradle task.

**Comment by qwazer** [comment]
Thanks for contributing @jigga and @imanushin 
FYI
Looks like gradle team change process of publishing.
Now they are review plugins before publishing.
I started publishing of v0.10.0

Response from Gradle:
> Publishing plugin com.github.qwazer.markdown-confluence version 0.10.0
> Thank you. Your new plugin com.github.qwazer.markdown-confluence has been submitted for approval by Gradle engineers. The request should be processed within the next few days, at which point you will be contacted via email.

**Pull request merged**